certora/specs/SocialRecoveryModule.spec

```
@@ -502,3 +503,113 @@ rule finalizeRecoveryAlwaysPossible(env e) {
+        f.selector == sig:finalizeRecovery(address).selector ||
+        f.selector == sig:cancelRecovery().selector;
+}
+
+// This rule verifies that is the confirmedHashes change, it must be one of the following functions:
+// - confirmRecovery(...)
+// - multiConfirmRecovery(...)
+rule confirmedHashesChange(env e, calldataarg args, method f, bytes32 hash, address guardian) {
+    bool confirmedHashBefore = currentContract.confirmedHashes[hash][guardian];
+
+    f(e, args);
+
+    bool confirmedHashAfter = currentContract.confirmedHashes[hash][guardian];
+
+    assert confirmedHashBefore != confirmedHashAfter =>
+        f.selector == sig:confirmRecovery(address,address[],uint256,bool).selector ||
+        f.selector == sig:multiConfirmRecovery(address,address[],uint256,SocialRecoveryModule.SignatureData[],bool).selector;
+}
+
+// This rule verifies that is the walletsNonces change, it must be one of the following functions:
+// - confirmRecovery(...)
+// - multiConfirmRecovery(...)
+// - executeRecovery(...)
+// - invalidateNonce(...)
+rule walletsNoncesChange(env e, calldataarg args, method f) {
+    uint256 walletsNoncesBefore = currentContract.walletsNonces[safeContract];
+
+    f(e, args);
+
+    uint256 walletsNoncesAfter = currentContract.walletsNonces[safeContract];
+
+    assert walletsNoncesBefore != walletsNoncesAfter =>
+        f.selector == sig:confirmRecovery(address,address[],uint256,bool).selector ||
+        f.selector == sig:multiConfirmRecovery(address,address[],uint256,SocialRecoveryModule.SignatureData[],bool).selector ||
+        f.selector == sig:executeRecovery(address,address[],uint256).selector ||
+        f.selector == sig:invalidateNonce().selector;
+}
+
+// This rule verifies that the recovery period never changes.
+rule recoveryPeriodNeverChange(env e, calldataarg args, method f) {
+    uint256 recoveryPeriodBefore = currentContract.recoveryPeriod;
+
+    f(e, args);
+
+    uint256 recoveryPeriodAfter = currentContract.recoveryPeriod;
+
+    assert recoveryPeriodBefore == recoveryPeriodAfter;
+}
+
+// This rule verifies that the guardian data can only be changed by the following functions:
+// - addGuardianWithThreshold(...)
+// - revokeGuardianWithThreshold(...)
+// - changeThreshold(...)
+rule entriesChange(env e, calldataarg args, method f) {
+    bytes32 guardiansHashBefore = currentContract.guardiansHash(safeContract);
+    uint256 guardiansCountBefore = currentContract.guardiansCount(safeContract);
+    uint256 guardianThresholdBefore = currentContract.threshold(safeContract);
+
+    f(e, args);
+
+    bytes32 guardiansHashAfter = currentContract.guardiansHash(safeContract);
+    uint256 guardiansCountAfter = currentContract.guardiansCount(safeContract);
+    uint256 guardianThresholdAfter = currentContract.threshold(safeContract);
+
+    assert (
+        guardiansHashBefore != guardiansHashAfter ||
+        guardiansCountBefore != guardiansCountAfter
+    ) =>
+    (
```

nit: for other things, we are using the style where the opening brace is on the previous line, I would use that here too:

```suggestion
    ) => (
```

---

Also, is the `()` after the `=>` needed? We don't use it in other places and it would be nice to have consistent formatting.

---

Updated 👍🏾